### PR TITLE
Bedrock: Add invite endpoints

### PR DESCRIPTION
**Pull request by LucienHH**

### CODE REVIEW

**Comment by LucienHH** [comment]
@extremeheat Once this has gone through review I'll add the further needed logic of accepting invites to bedrock-protocol as I feel like we shouldn't silently add a Realm If using `getRealmFromInvite`

**Comment by LucienHH** [comment]
You enter the Realms code, it fetches the Realm from that code and then you confirm if you want to add it or not

**Comment by extremeheat** [comment]
Yes so that should be the behavior here also. The goal here is not to just do a web request or something, but provide a high level API without all the nuances of their underlying API.

**Comment by extremeheat** [comment]
There can be more fine grained platform specific parts of the API to be clear. Just we should primarily be exposing high level methods analogous to in game behavior independent of how their API works for stability purposes. I don't know the specifics of this but it sounds like 1) the invite method needs to do the invite if needed or 2) add a new method that does this step, and publicly document this one.

**Comment by LucienHH** [comment]
I see. Currently, in this PR to replicate the client, it would look like this:
```js
const api = RealmAPI(authflow, 'bedrock')

const realm = await api.getRealmFromInvite(inviteCode)

if (!realm.member) await api.acceptRealmInviteFromCode(inviteCode)

const { ip, port } = await realm.getAddress()
```
But I guess the direction we're going in would be to handle that small part within `getRealmFromInvite`?

**Comment by extremeheat** [comment]
Yes there or in a separate function. There can be a function argument to skip the invite accept if needed, at which point you'd get an error from server or some similar response

File: src/bedrock/api.js [hunk 1]
```diff
@@ -11,13 +11,75 @@ module.exports = class BedrockRealmAPI extends RealmAPI {
     return { host, port: Number(port) }
   }
 
-  async getRealmFromInvite (realmInviteCode) {
+  async getRealmFromInvite (realmInviteCode, skipAccept = false) {
```
Comment (extremeheat) [inline src/bedrock/api.js:14]: You can also set this parameter to be invite=true. I think this would be clearer.

File: src/bedrock/api.js [hunk 2]
```diff
@@ -11,13 +11,75 @@ module.exports = class BedrockRealmAPI extends RealmAPI {
...
+        createdOn: invite.date
+      }
+    })
+  }
+
+  async acceptRealmInvitation (invitationId) {
+    await this.rest.put(`/invites/accept/${invitationId}`)
+  }
+
+  async rejectRealmInvitation (invitationId) {
```
Comment (extremeheat) [inline src/bedrock/api.js:73]: What's the difference between invitation ID and code?

Comment (LucienHH) [reply src/bedrock/api.js:73]: A Realm can have a single universal code which can be used by any player to add that Realm to their list.

You can also invite players individually, the invitationId is the id of that invite which you use to accept/reject

**Review by extremeheat — APPROVED** [review]

**Comment by extremeheat** [comment]
Have you tested this to work in game (if you are able to)? Lgtm for merge otherwise

**Comment by LucienHH** [comment]
Yeah, tested all the new endpoints and each worked as expected.

**Comment by LucienHH** [comment]
Can this be merged @extremeheat

**Comment by extremeheat** [comment]
Done, can you update the doc to separate out the bedrock API specific things to a separate md file?

**Comment by LucienHH** [comment]
Yeah plan to, https://github.com/PrismarineJS/prismarine-realms/issues/19

**Pull request merged**